src/router.js

```diff
@@ -101,8 +101,22 @@ const routes = [
   },
 ];
 
+const getInitialPath = () => {
+  if (window.location.hash.startsWith('#/')) {
+    const path = window.location.hash.replace('#/', '');
+    window.location.hash = '';
+    return path;
+  }
+  return null;
+};
+
+const initialPath = getInitialPath();
+if (initialPath) {
+  window.history.replaceState({}, '', initialPath);
```

Comment:
This ensures the initial path is set before the router is initialized. It limits the initial flash of one page then another, and avoids the need for any sort of router push. It will just read the right URL the first time.